### PR TITLE
Saves the settings for countdown-controls

### PR DESCRIPTION
**Pull request by GabrielInTheWorld**

### CODE REVIEW

File: client/src/app/core/ui-services/projection-dialog.service.ts [hunk 1]
```diff
@@ -55,10 +55,12 @@ export class ProjectionDialogService {
             const [action, projectors, projectorElement]: ProjectionDialogReturnType = response;
             if (action === 'project') {
                 this.projectorService.projectOnMultiple(projectors, projectorElement);
+                return { fullscreen: projectorElement.fullscreen, displayType: projectorElement.displayType };
             } else if (action === 'addToPreview') {
                 projectors.forEach(projector => {
                     this.projectorService.addElementToPreview(projector, projectorElement);
                 });
+                return null;
```
Comment (tsiegleauq) [inline client/src/app/core/ui-services/projection-dialog.service.ts]: return empty object *or* add null as returnable value
`: Promise<object | null>` or `: Promise<object> | null` (minor, fix later)

File: client/src/app/site/projector/components/countdown-controls/countdown-controls.component.ts [hunk 2]
```diff
@@ -110,7 +117,9 @@ export class CountdownControlsComponent extends BaseViewComponent {
      * Brings the projection dialog
      */
     public onBringDialog(): void {
```
Comment (FinnStutzenstein) [inline client/src/app/site/projector/components/countdown-controls/countdown-controls.component.ts]: Please use an async function here.

File: client/src/app/site/projector/components/countdown-controls/countdown-controls.component.ts [hunk 3]
```diff
@@ -123,4 +132,13 @@ export class CountdownControlsComponent extends BaseViewComponent {
...
         }
     }
+
+    /**
+     * Stores the options for a projector in the local-storage.
+     *
+     * @param element The configured options for projector
+     */
+    private storeSettings(element: object): void {
+        this.storage.set(this.storageKey, element);
```
Comment (FinnStutzenstein) [inline client/src/app/site/projector/components/countdown-controls/countdown-controls.component.ts]: Please await this.

File: client/src/app/shared/components/projector-button/projector-button.component.ts [hunk 4]
```diff
@@ -115,8 +117,11 @@ export class ProjectorButtonComponent implements OnInit, OnDestroy {
                     // remove the projected object
                     this.projectorService.removeFrom(this.projector, this.object);
                 } else {
+                    const projectorElement = this.getProjectorElement(
+                        await this.storage.get<object>('projectorElementOptions')
+                    );
                     // instantly project the object
-                    this.projectorService.projectOn(this.projector, this.object);
+                    this.projectorService.projectOn(this.projector, projectorElement || this.object);
```
Comment (FinnStutzenstein) [inline client/src/app/shared/components/projector-button/projector-button.component.ts]: please remove `|| this.object` and fix the `getProjectorElement` function

File: client/src/app/site/projector/components/countdown-controls/countdown-controls.component.ts [hunk 5]
```diff
@@ -44,6 +45,11 @@ export class CountdownControlsComponent extends BaseViewComponent {
      */
     public warningTime: number;
 
+    /**
+     * The key to storage.
+     */
+    private storageKey = 'projectorElementOptions';
```
Comment (FinnStutzenstein) [inline client/src/app/site/projector/components/countdown-controls/countdown-controls.component.ts]: The value `projectorElementOptions` is used twice in this PR. Please put it in a constant above the class, export it and reuse the constant.

File: client/src/app/core/ui-services/projection-dialog.service.ts [hunk 6]
```diff
@@ -55,10 +55,12 @@ export class ProjectionDialogService {
             const [action, projectors, projectorElement]: ProjectionDialogReturnType = response;
             if (action === 'project') {
                 this.projectorService.projectOnMultiple(projectors, projectorElement);
+                return { fullscreen: projectorElement.fullscreen, displayType: projectorElement.displayType };
```
Comment (FinnStutzenstein) [inline client/src/app/core/ui-services/projection-dialog.service.ts]: This is countdown related code!
Please make this more generic and extract all extra properties from the projector Element.

**Comment by FinnStutzenstein** [comment]
I'm not sure why this was merged so fast, but **please** rework this.

**Comment by tsiegleauq** [comment]
cause of high demand. There is no breaking change nor bug. @GabrielInTheWorld can adjust the code style later.

**Pull request merged**